staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -167,3 +167,52 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32 `json:"samplingRatePerMillion,omitempty" protobuf:"varint,2,opt,name=samplingRatePerMillion"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityAndFairnessConfiguration holds API Priority and Fairness parameters.
+type PriorityAndFairnessConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// WorkEstimator holds work estimator parameters.
+	// +optional
+	WorkEstimator WorkEstimatorConfiguration `json:"workEstimator,omitempty"`
+}
+
+// WorkEstimatorConfiguration holds work estimator parameters.
+type WorkEstimatorConfiguration struct {
+	// ListWorkEstimator holds work estimator parameters related to list requests.
+	// +optional
+	ListWorkEstimator ListWorkEstimatorConfiguration `json:"listRequests,omitempty"`
+	// MutatingWorkEstimator holds work estimator parameters
+	// related to requests that mutate watched objects.
+	MutatingWorkEstimator MutatingWorkEstimatorConfiguration `json:"mutatingRequests,omitempty"`
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	// +optional
+	MaximumSeats *uint64 `json:"maximumSeats,omitempty"`
+}
+
+// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfiguration struct {
+	// ObjectsPerSeat is a number of objects from a single list request per seat.
+	// +optional
+	ObjectsPerSeat *float64 `json:"objectsPerSeat,omitempty"`
+}
+
+// MutatingWorkEstimatorConfiguration holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfiguration struct {
+	// EventAdditionalDuration is the additional amount of time that a mutating request occupies seats
+	// to account for the work involved in sending the corresponding WATCH notifications. Setting its
+	// value to 0 effectively disables this behavior but a request may still occupy seats.
+	// +optional
+	EventAdditionalDuration *metav1.Duration `json:"eventAdditionalDuration,omitempty"`
+	// WatchesPerSeat is the ratio between the number of WATCHes that will get notified and
+	// the number of seats occupied for EventAdditionalDuration to account for the work of
+	// sending those notifications. Setting its value to 0 disables mutating work estimator.
```

Comment:
I am not a great fan of defining zero to have an irregular meaning.
There are one or two other ways to get the same effect: setting a high WatchesPerSeat will surely do it, and we can reasonably define EventAdditionalDuration=0 to also do it.

I see that the early-out for 0 here is earlier that the out for a high value of WatchesPerSeat, and that might conceptually be worth something.  But defining EventAdditionalDuration=0 to fully disable the extra delay would be more uniform semantically and cause the same early-out.